### ISSUE TITLE
ltr "programming error" with filter

### ISSUE THREAD

**Issue by AdamantConlanger** [def-gthill/lexurgy#56]
Minimal example:
```
Feature type(consonant, vowel)
Feature (syllable) stress(*unstressed, stress)

Symbol C [consonant]
Symbol V [vowel]

Diacritic ˈ (before) (floating) [stress]

Syllables:
  C? V C? C?

rule-one [vowel]:
  [] => [stress] / $ _

rule-two ltr [vowel]:
  [unstressed] => [stress] / [stress] [] _
```
Input: `CVCCVCVV`

**Comment by AdamantConlanger** [comment]
for future reference in case the title gets changed:
The problem occurs at rule-two, where it gives the error message "Rule rule-two encountered a programming error when applied to word ˈCVC.CV.CV.V (originally CVCCVCVV)". I've never encountered that error before (as far as I can recall; it's been a while.) Removing the left-to-rightness of the rule fixes the issue, as does removing the filter or just removing the entire rule. Interestingly, replacing ltr by propagate fixes it.

Also, the original minimal example doesn't trip up with input `CVCCVCV`, but if we make it rtl instead of ltr, it errors there too.

**Comment by def-gthill** [comment]
Good catch. A "programming error" means there was an error condition somewhere, but I haven't told the program what the error message should be for that condition. It's always a bug: either the sound changes should work, or the sound changes are invalid and you should get a real error message explaining why.

**Comment by def-gthill** [comment]
Fixed in commit 5753f3f1c9dc0ec9618b49c463936c28118be79e
